### DISCUSSION TITLE
Newbie getting my head around kickstarter setup

### DISCUSSION THREAD

**Discussion by BarbaraRobson** [quarto-dev/quarto-nvim#91] [Q&A] [answered]
I've just installed `quarto-nvim-kickstarter` on my Windows box and wow! Thank you for providing this -- there is so much useful functionality here! However, it is a lot all at once for someone who has come from years of using a relatively vanilla neovim setup with only `NVim-R` installed. I am having a few issues: some of these are probably Windows things, some might be because I've set it up on my work laptop that has some extra security hurdles to installations, and some of this is just because I don't know where everything is with all these packages that are new to me. I mostly have it working, but I'd appreciate any tips on any of the following teething problems:

- Each time I launch `nvim`, it tries unsuccessfully to install the Haskell language server, and I get an error message from `mason-lsp.config`. I don't use Haskell, so I don't know whether this is supposed installed in the default setup or whether I accidentally told Mason it should be installed when I was working on getting the python and R language servers working. Am I right in assuming hsl is not needed if I don't use Haskell? If so, which config file do I need to edit to tell it not to try installing it?
- `:PasteImg` doesn't work. The first time I try it in a new neovim window, I get an error message:
`E5108: Error executing lua ...lazy/clipboard-image.nvim/lua/clipboard-image/health.lua:3: module 'health' not found: no field package.preLoad['health']`
- The spellchecker mostly works but does not underline errors in red. Any tips?
- Some of the new keybindings are taking some getting used to. E.g., I am used to using 's' and '^B' for 'substitute' and 'back page' respectively. Where can I find and edit these key mappings and is there an alternative shortcut to jump back by a page?

**Comment by jmbuhr** [comment, marked as answer]
Some of your points are actual issues with the config, thank you for spotting! I'll address those in the next release.

- the haskell ls should indeed not be needed for most people and is an artifact of me using a dev branch of this as my personal config. This enables me to keep checking that stuff works together, but sometimes also leads to this. To disable it look for lspconfig in quarto.lua and remove the hls paragraph.
- the PasteImg plugin has been broken for a while now and I completely forget about looking for a replacement or fix. When it first broke I had to actually work (🙈), so I ended up using a terminal bash alias to paste an image from the clipboard and then include it manually. But I'll look into the plugin again!
- is the spell-checker enabled? It should be a rather vanilla nvim feature
- I tend to use ctrl+d/u to go down or up. The s shortcut comes from the flash.nvim plugin, maybe this also shouldn't be enabled by default.

The last point raises an interesting question as to how this config should be used. Since using quarto and data sciency things together requires a number of plugins (e.g. autocompletion, language server configuration, treesitter highlighting) I created this to show the plugin in its natural habitat. Everything that has a direct effect on your experience with a quarto document is in quarto.lua. This should make it easier for people to copy the relevant parts into their own configs into files they see fit. But there are also plugins to round things off that maybe you don't need.

**Reply by jmbuhr** [reply]
Here I have another example of including quarto features into an existing config based on LazyVim: https://github.com/jmbuhr/lazyvim-starter-for-quarto

In this case, the quarto.lua file is even smaller, because it only extends what is already in LazyVim, so maybe this makes it easier to see the direction relations.

**Reply by BarbaraRobson** [reply]
Thank you very much! I found the hsl line and disabled it, so that is sorted out now.
You were right -- I didn't have spellcheck enabled. 🤪
I do quite like flash.nvim now that I know what it is and where it lives, so I will keep it, but might change the shortcut.
I think might remap ctrl+u (or something else) to bring up nvim-tree and get ctrl+b back for backpage. I'm still having a bit of trouble finding the config file responsible for that mapping?

I think I do want all the packages in the kickstarter -- I just need to put together a cheatsheet so I know which package does what and what shortcuts are provided. Thanks again!

**Reply by jmbuhr** [reply]
some shortcuts are defined not on the keymaps.lua file, but directly in the plugin specification used by lazy.nvim to only load the plugin once one of its keys has been used. In ui.lua you'll find the spec for nvim-tree, including the key :)
